subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/processors/RestartEveryNTestClassProcessor.java

```diff
@@ -66,8 +66,9 @@ public void stop() {
     @Override
     public void stopNow() {
         stoppedNow = true;
-        if (processor != null) {
-            processor.stopNow();
+        TestClassProcessor toStop = processor;
```

Comment:
This non-atomic check-then-act risk can also happen in `stop` and `processTestClass` method. Let's fix all of them to avoid rework future.